meerk40t/gui/scene/scene.py

```diff
@@ -278,13 +278,15 @@ def on_right_mouse_up(self, event):
 
     def on_magnify_mouse(self, event):
         """
-        Magnify Mouse is a Mac Event called with pinch to zoom on a trackpad.
+        Magnify Mouse is a Mac-only Event called with pinch to zoom on a trackpad.
         """
-        magnify = event.GetMagnification()
-        if magnify > 0:
-            self.scene.event(event.GetPosition(), "zoom-in")
-        if magnify < 0:
-            self.scene.event(event.GetPosition(), "zoom-out")
+        magnification = event.GetMagnification()
+        if event.IsGestureStart():
+            self.scene.event(event.GetPosition(), "gesture-start")
+        elif event.IsGestureEnd():
+            self.scene.event(event.GetPosition(), "gesture-end")
+        else:
+            self.scene.event(event.GetPosition(), "zoom {mag}",format(mag=magnification))
```

Comment:
This can't be used. "zoom " here has been used for the pinch and zoom screen gestures which are already using that functionality and that zoom is active from gesture-start.

1) Touch Screen Pinch-Zoom: ZoomFactor is zoom from the start of the zoom event.
2) TouchPad Pinch-Zoom: Magnify is the instantaneous zoom from the previous zoom event in a range of around [-0.4,0.4].

These two are not compatible. One is compounded the other is not.